internal/pkg/fieldpropagator/analyzer.go

```diff
@@ -44,40 +45,58 @@ var Analyzer = &analysis.Analyzer{
 A field propagator is a function that returns a source field.`,
 	Flags:      config.FlagSet,
 	Run:        run,
-	Requires:   []*analysis.Analyzer{buildssa.Analyzer},
+	Requires:   []*analysis.Analyzer{buildssa.Analyzer, fieldtags.Analyzer},
 	ResultType: reflect.TypeOf(new(ResultType)).Elem(),
 	FactTypes:  []analysis.Fact{new(isFieldPropagator)},
 }
 
+type Analysis struct {
```

Comment:
WDYT about this? I find it makes the function signatures much lighter, and the code more readable.

The name might not be ideal though, I am open to suggestions.

Reply:
Hmm. I'm not sure I see any clear drawbacks here. The abstraction would be the state/functionality that is necessary to run the analysis itself. A 4-argument signature is never fun to see in any case.

It is indeed not meant to be exported, good catch.

I don't see how this hurts the reviewability of this PR, and I don't think this change would hold its own in a future PR. I made this change because I thought it was justified by the changes in the PR, i.e. it got to a point where we had a bunch of functions with large signatures, and I was looking for a way to make a clear distinction between "these variables are specifically important to this function" and "these variables are generally important to the analysis as a whole".

Reply:
> A 4-argument signature is never fun to see in any case.
> ...
> I was looking for a way to make a clear distinction between "these variables are specifically important to this function" and "these variables are generally important to the analysis as a whole".

This isn't Java.  Go is a functional language, not an object-oriented one.  Structs are abstracted data bundles, not namespaces.  There's no benefit in calling a niladic function over a four-parameter one.  A function should have as many arguments as it needs and not rely on a struct to globalize parameters.

> I don't see how this hurts the reviewability of this PR

As your reviewer, I am letting you know that it has slowed my review.

> I don't think this change would hold its own in a future PR.

There's no `You must be this tall to open a PR` here.  If it is a meaningful, atomic improvement, it could constitute a quick-and-easy PR.  If it isn't, it doesn't belong here either.

Reply:
> This isn't Java. 

I don't see the point in comparing Go to other languages, or in discussing programming paradigms. My understanding of your point of view is that the `struct` keyword should never be used unless it is defining a meaningful abstraction. Is that correct?

> There's no benefit in calling a niladic function over a four-parameter one.

Let's agree to disagree then. I think a function is much easier to read when it has fewer parameters.

> A function should have as many arguments as it needs ...

So in your opinion a function with, say, 8 arguments would not be a problem?

> As your reviewer, I am letting you know that it has slowed my review.

Gotcha. I will try to avoid creating unnecessary structs. 

> There's no ...

Ok, that is good to know. To reiterate my initial stance, I thought the change was warranted as a natural extension of the changes in this PR. It certainly wasn't *required* by the changes though, so I guess it could indeed have been its own PR.